logql/v2/parser_test.go

```diff
@@ -385,6 +386,36 @@ func TestParseExpr(t *testing.T) {
 				},
 			},
 		},
+		{
+			input: `{first="value"} | second=~"foo|bar" or third=~"foo|bar"`,
+			expr: &LogQueryExpr{
+				filter: LogPipelineExpr{
+					&LogLabelFilterExpr{
+						labelName:    "second",
+						comparisonOp: "=~",
+						labelValue:   "foo|bar",
+						right: []*LogLabelFilterExpr{
+							{
+								labelName:    "third",
+								comparisonOp: "=~",
+								labelValue:   "foo|bar",
+								isNested:     true,
+								chainOp:      "or",
+							},
+						},
+					},
+				},
+				left: &StreamMatcherExpr{
+					matchers: []*labels.Matcher{
+						{
+							Type:  labels.MatchEqual,
+							Name:  "first",
+							Value: "value",
+						},
+					},
+				},
+			},
+		},
```

Comment:
Isn't this test case already covered by the next one? Or is this all about testing the `strconv.Quote`? If yes, on the last question then we can drop this test as per `strconv.Quote` having enough tests in the Golang std library.

Reply:
I think I wrote this test because it was the closest I had from the actual netobserv queries with tenancy matchers. So it helped me while doing the implementation. But you're right that it overlaps with other tests. I can remove it.